x/concentrated-liquidity/client/cli/tx.go

```diff
@@ -76,3 +86,115 @@ func NewCreateIncentiveCmd() (*osmocli.TxCliDesc, *types.MsgCreateIncentive) {
 		Flags:               osmocli.FlagDesc{RequiredFlags: []*flag.FlagSet{FlagSetJustPoolId()}},
 	}, &types.MsgCreateIncentive{}
 }
+
+func NewTickSpacingDecreaseProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tick-spacing-decrease-proposal [flags]",
+		Args:  cobra.ExactArgs(0),
+		Short: "Submit a tick spacing decrease proposal",
+		Long: strings.TrimSpace(`Submit a tick spacing decrease proposal.
+
+Passing in FlagPoolIdToTickSpacingRecords separated by commas would be parsed automatically to pairs of PoolIdToTickSpacing records.
+Ex) --pool-tick-spacing-records=1,10,5,1 -> [(poolId 1, newTickSpacing 10), (poolId 5, newTickSpacing 1)]
+Note: The new tick spacing value must be less than the current tick spacing value.
+
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			content, err := parsePoolIdToTickSpacingRecordsArgsToContent(cmd)
+			if err != nil {
+				return err
+			}
+
+			from := clientCtx.GetFromAddress()
+
+			depositStr, err := cmd.Flags().GetString(govcli.FlagDeposit)
+			if err != nil {
+				return err
+			}
+			deposit, err := sdk.ParseCoinsNormalized(depositStr)
+			if err != nil {
+				return err
+			}
+
+			msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+			if err != nil {
+				return err
+			}
+
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	cmd.Flags().String(govcli.FlagTitle, "", "title of proposal")
+	cmd.Flags().String(govcli.FlagDescription, "", "description of proposal")
+	cmd.Flags().String(govcli.FlagDeposit, "", "deposit of proposal")
+	cmd.Flags().Bool(govcli.FlagIsExpedited, false, "If true, makes the proposal an expedited one")
+	cmd.Flags().String(govcli.FlagProposal, "", "Proposal file path (if this path is given, other proposal flags are ignored)")
+	cmd.Flags().String(FlagPoolIdToTickSpacingRecords, "", "The pool ID to new tick spacing records array")
+
+	return cmd
+}
+
+func parsePoolIdToTickSpacingRecordsArgsToContent(cmd *cobra.Command) (govtypes.Content, error) {
+	title, err := cmd.Flags().GetString(govcli.FlagTitle)
+	if err != nil {
+		return nil, err
+	}
+
+	description, err := cmd.Flags().GetString(govcli.FlagDescription)
+	if err != nil {
+		return nil, err
+	}
+
+	poolIdToTickSpacingRecords, err := parsePoolIdToTickSpacingRecords(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	content := &types.TickSpacingDecreaseProposal{
+		Title:                      title,
+		Description:                description,
+		PoolIdToTickSpacingRecords: poolIdToTickSpacingRecords,
+	}
+	return content, nil
+}
+
+func parsePoolIdToTickSpacingRecords(cmd *cobra.Command) ([]types.PoolIdToTickSpacingRecord, error) {
+	assetsStr, err := cmd.Flags().GetString(FlagPoolIdToTickSpacingRecords)
+	if err != nil {
+		return nil, err
```

Comment:
can we add a check to see if the len(assets) is even number?

Reply:
Makes sense to me, added here https://github.com/osmosis-labs/osmosis/pull/4982/commits/54103588e1ad7c771a7b6532ca9bebfea7eca607